Student/StudentUITests/Tests/AssignmentDetailsPageTest.swift

```diff
@@ -24,27 +24,31 @@ class AssignmentDetailsPageTest: StudentTest {
     let submissionDetailsPage = SubmissionDetailsPage.self
 
     lazy var course: APICourse = {
-        return seedClient.createCourse()
-    }()
-    lazy var teacher: AuthUser = {
-        return createTeacher(in: course)
-    }()
-    lazy var student: AuthUser = {
-        return createStudent(in: course)
+        let course = APICourse.make()
+        mockData(GetCourseRequest(courseID: course.id), value: course)
+        return course
     }()
 
-    func testUnsubmittedUpload() {
-        seedClient.updateCustomColor(user: student, context: ContextModel(.course, id: course.id), hexcode: "123456")
-        let assignment = seedClient.createAssignment(
-            for: course,
-            description: "A description",
-            pointsPossible: 12.3,
-            dueAt: DateComponents(calendar: Calendar.current, year: 2035, month: 1, day: 1, hour: 8).date,
-            submissionTypes: [.online_upload],
-            allowedExtensions: ["doc", "docx", "pdf"]
-        )
-        launch("/courses/\(course.id)/assignments/\(assignment.id)", as: student)
+    func mockAssignment(_ assignment: APIAssignment) -> APIAssignment {
+        mockData(GetAssignmentRequest(courseID: course.id, assignmentID: assignment.id.value, include: [.submission]), value: assignment)
+        return assignment
+    }
 
+    func testUnsubmittedUpload() {
+        mockData(GetCustomColorsRequest(), value: APICustomColors(custom_colors: [
+            course.canvasContextID: "#123456",
+        ]))
+        host.logIn(domain: "canvas.instructure.com", token: "")
```

Comment:
What is this?

Reply:
It's in the login flow that the custom colors get requested, this triggers that flow so we can check navigation color.